tests/ui/job-view/details/PinBoard_test.jsx

```diff
@@ -0,0 +1,171 @@
+import { getApiUrl } from '../../../../ui/helpers/url';
+import FilterModel from '../../../../ui/models/filter';
+import {
+  replaceLocation,
+  getProjectUrl,
+} from '../../../../ui/helpers/location';
+import configureStore from '../../../../ui/job-view/redux/configureStore';
+import { setSelectedJob } from '../../../../ui/job-view/redux/stores/selectedJob';
+import { setPushes } from '../../../../ui/job-view/redux/stores/pushes';
+import reposFixture from '../../mock/repositories';
+import KeyboardShortcuts from '../../../../ui/job-view/KeyboardShortcuts';
+import { pinJobs } from '../../../../ui/job-view/redux/stores/pinnedJobs';
+
+describe('DetailsPanel', () => {
+  const repoName = 'autoland';
+  const classificationTypes = [{ id: 1, name: 'intermittent' }];
+  const classificationMap = { 1: 'intermittent' };
+  const filterModel = new FilterModel();
+  let jobList = null;
+  const currentRepo = reposFixture[2];
+  currentRepo.getRevisionHref = () => 'foo';
+  currentRepo.getPushLogHref = () => 'foo';
+
+  beforeEach(async () => {
+    fetchMock.get(
+      getApiUrl('/jobs/?push_id=511138', repoName),
+      jobListFixtureOne,
+    );
+    jobList = await JobModel.getList({ push_id: 511138 });
+
+    fetchMock.get(getProjectUrl('/jobs/259537372/', repoName), jobList.data[1]);
+    fetchMock.get(getProjectUrl('/note/?job_id=259537372', repoName), []);
+    fetchMock.get(
+      getProjectUrl('/jobs/259537372/text_log_steps/', repoName),
+      [],
+    );
+    fetchMock.get(
+      getProjectUrl('/jobs/259537372/bug_suggestions/', repoName),
+      [],
+    );
+    fetchMock.get(
+      getProjectUrl('/bug-job-map/?job_id=259537372', repoName),
+      [],
+    );
+    fetchMock.get(
+      getProjectUrl('/performance/data/?job_id=259537372', repoName),
+      [],
+    );
+    fetchMock.get(
+      getProjectUrl('/job-log-url/?job_id=259537372', 'autoland'),
+      [],
+    );
+    fetchMock.get(getApiUrl('/jobdetail/?job_id=259537372'), { results: [] });
+  });
+
+  afterEach(() => {
+    cleanup();
+    fetchMock.reset();
+    replaceLocation({});
+  });
+
+  const testDetailsPanel = store => (
+    <div id="global-container" className="height-minus-navbars">
+      <Provider store={store}>
+        <KeyboardShortcuts
+          filterModel={filterModel}
+          showOnScreenShortcuts={() => {}}
+        >
+          <div />
+          <div id="th-global-content" data-testid="global-content">
+            <DetailsPanel
+              currentRepo={currentRepo}
+              user={{ isLoggedIn: false }}
+              resizedHeight={100}
+              classificationTypes={classificationTypes}
+              classificationMap={classificationMap}
+            />
+          </div>
+        </KeyboardShortcuts>
+      </Provider>
+    </div>
+  );
+
+  test('pin selected job with button', async () => {
+    const { store } = configureStore();
+    store.dispatch(setPushes(pushFixture.results, {}));
```

Comment:
Does it make sense to move these two lines to `beforeEach`? (and the same lines on the other tests)

Reply:
Good call!  Done...